go/runtime/scheduling/simple/txpool/priorityqueue/priority_queue.go

```diff
@@ -149,6 +149,24 @@ func (q *priorityQueue) GetBatch(force bool) []*transaction.CheckedTransaction {
 	return batch
 }
 
+// Implements api.TxPool.
+func (q *priorityQueue) GetKnownBatch(batch []hash.Hash) ([]*transaction.CheckedTransaction, map[hash.Hash]int) {
+	q.Lock()
+	defer q.Unlock()
+
+	var result []*transaction.CheckedTransaction
```

Comment:
`result` could be pre-allocated since the size is known to be `len(batch)`

Reply:
Done.